functions/pyenv-init.fish

```diff
@@ -0,0 +1,16 @@
+
+function pyenv-init --description "Pyenv initialization script for fish shell"
+
+    # Checks if directories already exists. If not, creates then
+    for dir in "shims" "bin"
+        if not test -d $HOME/pyenv/$dir
+            mkdir -p $HOME/pyenv/$dir
+        end
+    end
+
+    # Respects if PYENV_ROOT exists. Otherwise, sets it to a default directory
+    set --query PYENV_ROOT; or set --local PYENV_ROOT $HOME/.pyenv
+
+    # Appends PYENV_ROOT directories to PATH variable
+    set PATH $PYENV_ROOT/shims $PYENV_ROOT/bin $PATH
```

Comment:
Looks like this might be something related to making `pyenv` command available, but my experience tells me `shims` suffices. Not sure how it works on OSs other than OS X thou.